smolyay/samples.py

```diff
@@ -0,0 +1,362 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points to be used as the sampling points
+    in a tensor product grid or sparse grid. These unidimensional points are
+    associated with certain polynomial families, and the combination of
+    unique points and polynomial families are the basis for different types
+    of quadrature rules that are used in numerical integration and in
+    approximation.
+    The natural domain of the points specified by :param:``domain`` is
+    the domain in which all the possible points in the set lies within.
+    ``points`` is the list of points that the UnidimensionalPointSet
+    is expected to store.
+    ``domain`` is the domain of the points generated by the point set.
+    """
+
+    def __init__(self):
```

Comment:
I thought that these point sets would take a domain, and deriving classes would scale the points as needed? You will need to add an argument here for the domain, coerce it to a numpy array, and add a readonly property.

Reply:
I was under the impression that it was the MultidimensionalPointSet that would be handling the domain and scaling.

Reply:
No, that won’t work with the interface we discussed with Dr. K at our last meeting to build a multidimensional set from a list of unidimensional sets. The unidimensional sets need to already have their domains set for that to work.

Reply:
Okay, I went ahead and added domain as a parameter.